sig-scalability/goals.md

```diff
@@ -0,0 +1,78 @@
+* Max machines per cluster
+  * Goal: 5,000
+  * This is somewhat greater than (max cores per cluster) / (max cores per node), but not excessively so.  The primary metric is cores/cluster, but we do not believe that it is worth supporting the maximum core count per cluster on exclusively small (e.g. only single-core) machines. Conversely, we do not expect users to only be able to achieve the max cores per cluster limit by using the largest commercially available machines.  Using current numbers however, we plan to support clusters of e.g. 5,000 x 32 core machines = 160,000 cores.
+* Max pods per cluster
+  * Goal: 500,000
+  * Note: this is somewhat less than (max cores per cluster) x (max pods per core).  I.e. The largest possible cluster, packed exclusively with the smallest possible pods, is not supported, as we have not (yet) found a good use case for this (see also Open Questions below).
+* End-to-end pod startup time
+  * Goal: <= 5 seconds (99th percentile)
+  * With pre-pulled images, saturated cluster, no scheduling backlog.
+* Scheduler throughput
+  * Goal: 100 pods per second
+  * Driven primarily by max pods per cluster, and max cluster saturation time (see below).
+* Max cluster saturation time
+  * Goal: 90 minutes
+  * Time to restore a fully saturated large cluster is important for cluster-wide failure recovery, and/or related emergency capacity provisioning (e.g. building and populating a new cluster to replace capacity in a failed one). This number also needs to correlate with max pods per cluster, and max scheduler throughput (500,000 pods / 100 pods per second ~ 90 minutes).  We believe that this fulfills most real-world recovery requirements.  The required time to recovery is usually driven primarily by trying to reduce the probability of multiple uncorrelated cluster failures (e.g. "one of our 3 clusters has failed. We're just fine unless another one fails before we've repaired/replaced the first failed one").
+
+## Open Questions
+
+1. **What, if any, reasonable use cases exist for very large numbers of very small nodes (e.g. for isolation reasons - multitenant)?  Based on comments so far, it seems that the answer is yes, and needs to be addressed.**<br>
+The above scaling goals explicitly accommodate a maximum of 5,000 nodes.  Do we need a special case for larger numbers of small nodes (e.g. 200,000  single-core nodes).  The latter already fits within our other scalability limits (cores per cluster, pods per cluster), so it might not be more difficult to achieve than those.  Two example use cases I've heard anecdotally are (a) running e.g. large numbers of customers' small, largely idle wordpress instances, one per node, and (b) giving away limited numbers of free
```

Comment:
Truncated sentence: "limited numbers of free".. what? Do you remember @jbeda ?

Reply:
See the original doc from where it was copied:
https://docs.google.com/document/d/1pABjn69yeDJDl0Yrsneiwj5A_bUI7BZgvKTQtSwzjiE/edit#

Reply:
Yes, it looks like a cut 'n paste error happened here.  Probably worth fixing.